lua/line-number-change-mode/init.lua

```diff
@@ -4,26 +4,30 @@ function M.setup(opts)
    opts = opts or {}
 
    local group = vim.api.nvim_create_augroup("LineNumberChangeMode", {
-      clear = true
+      clear = true,
    })
 
-   vim.api.nvim_create_autocmd('ModeChanged', {
-      group = group,
-      callback = function()
-         local new_mode = vim.v.event.new_mode
-         local line_number_map = {
-            i = opts.i and opts.i or nil,
-            n = opts.n and opts.n or nil,
-            no = opts.no and opts.no or nil,
-            R = opts.R and opts.R or nil,
-            v = opts.n and opts.v or nil,
-            V = opts.V and opts.V or nil,
-         }
-
-         if line_number_map[new_mode] ~= nil and opts[new_mode] ~= nil then
-            vim.api.nvim_set_hl(0, 'CursorLineNr', line_number_map[new_mode])
+   local line_number_map = opts
+
+   local function set_hl_for_mode(mode)
+      if line_number_map[mode] then
+         vim.api.nvim_set_hl(0, "CursorLineNr", line_number_map[mode])
+
+         -- Have to force a redraw to repaint line number if switching to cmd mode
+         if mode == "c" then
+            vim.cmd.redraw()
          end
       end
+   end
+
+   -- set mode on setup to set CursorLineNr in case it's not already set
+   set_hl_for_mode(vim.api.nvim_get_mode().mode)
```

Comment:
Would loading Neovim in the beginning always be in normal mode?  Seems like we could set some sensible defaults instead.

Reply:
Sorry, I don't think my comment wasn't clear. 

On startup, if someone's CursorLineNr highlight for normal mode isn't the same as set in this plugin, it'll be a different color until they switch to another mode and back to normal mode because there's no ModeChanged event on startup. If the CursorLineNr highlight is the same as configured in the plugin, then it wouldn't matter.

I think some (most?) folks will just want to configure CursoreLineNr here and not have change the default in their theme so this code makes sure to set CursorLineNr on startup (or whenever the plugin is loaded)